### ISSUE TITLE
replace has problems with empty strings

### ISSUE THREAD

**Issue by fulara** [rust-lang/regex#394]
Seems like a replace has problems processing empty strings. example:
```
extern crate regex;
use regex::Regex;
fn main() {
    let regex = Regex::new("\\d*").unwrap();
    let s = regex.replace("", "replaced").into_owned();
    assert_eq!("replaced", s);
}
```
That fails. however:
```
extern crate regex;
use regex::Regex;
fn main() {
    let regex = Regex::new("\\d*").unwrap();
    let s = regex.find("");
    assert!(s.is_some());
}
```
That one passes.
Thanks to lapinot I've workaround here:
```
extern crate regex;
use regex::Regex;
fn main() {
    let regex = Regex::new("\\d*").unwrap();
    let s = regex.replace("", |caps: &regex::Captures| { "replaced".to_string() }).into_owned();
    assert_eq!("replaced", s);
}
```

**Comment by BurntSushi** [comment]
This is probably a dupe of #393, but we can leave this open until a fix confirms.

**Comment by BurntSushi** [comment]
Fixed by #395
